go/vt/sqlparser/sql_test.go

```diff
@@ -280,6 +280,264 @@ func TestShowIndex(t *testing.T) {
+						ReferencedColumns: Columns{{val: "id"}, {val: "sub_id"}},
+						OnUpdate: DefaultAction,
+						OnDelete: DefaultAction,
+					},
+				},
+			},
+		},
+	}
+
+	testForeignKey(t, tests)
+}
+
+func TestAlterAddForeignKey(t *testing.T) {
+	tests := []testForeignKeyStruct{
+		{
+			`ALTER TABLE child ADD
+				FOREIGN KEY (parent_id)
+					REFERENCES parent(id)
+					ON UPDATE RESTRICT ON DELETE CASCADE`,
+			[]*ConstraintDefinition{
+				{
+					Name: "",
+					Details: &ForeignKeyDefinition{
+						IsDrop: false,
+						Source: Columns{{val: "parent_id"}},
+						ReferencedTable: TableName{Name: TableIdent{v: "parent"}},
+						ReferencedColumns: Columns{{val: "id"}},
+						OnUpdate: Restrict,
+						OnDelete: Cascade,
+					},
+				},
+			},
+		},
+		{
+			`ALTER TABLE child ADD
+				FOREIGN KEY (parent_id)
+					REFERENCES parent(id)
+					ON DELETE NO ACTION ON UPDATE SET DEFAULT`,
+			[]*ConstraintDefinition{
+				{
+					Name: "",
+					Details: &ForeignKeyDefinition{
+						IsDrop: false,
+						Source: Columns{{val: "parent_id"}},
+						ReferencedTable: TableName{Name: TableIdent{v: "parent"}},
+						ReferencedColumns: Columns{{val: "id"}},
+						OnUpdate: SetDefault,
+						OnDelete: NoAction,
+					},
+				},
+			},
+		},
+		{
+			`ALTER TABLE child ADD
+				FOREIGN KEY (parent_id)
+					REFERENCES parent(id)
+					ON DELETE CASCADE`,
+			[]*ConstraintDefinition{
+				{
+					Name: "",
+					Details: &ForeignKeyDefinition{
+						IsDrop: false,
+						Source: Columns{{val: "parent_id"}},
+						ReferencedTable: TableName{Name: TableIdent{v: "parent"}},
+						ReferencedColumns: Columns{{val: "id"}},
+						OnUpdate: DefaultAction,
+						OnDelete: Cascade,
+					},
+				},
+			},
+		},
+		{
+			`ALTER TABLE child ADD
+				FOREIGN KEY (parent_id)
+					REFERENCES parent(id)
+					ON UPDATE SET NULL`,
+			[]*ConstraintDefinition{
+				{
+					Name: "",
+					Details: &ForeignKeyDefinition{
+						IsDrop: false,
+						Source: Columns{{val: "parent_id"}},
+						ReferencedTable: TableName{Name: TableIdent{v: "parent"}},
+						ReferencedColumns: Columns{{val: "id"}},
+						OnUpdate: SetNull,
+						OnDelete: DefaultAction,
+					},
+				},
+			},
+		},
+		{
+			`ALTER TABLE child ADD
+				CONSTRAINT fk_parents FOREIGN KEY (parents_id, parents_sub_id)
+					REFERENCES parents(id, sub_id)`,
+			[]*ConstraintDefinition{
+				{
+					Name: "fk_parents",
+					Details: &ForeignKeyDefinition{
+						IsDrop: false,
+						Source: Columns{{val: "parents_id"}, {val: "parents_sub_id"}},
+						ReferencedTable: TableName{Name: TableIdent{v: "parents"}},
+						ReferencedColumns: Columns{{val: "id"}, {val: "sub_id"}},
+						OnUpdate: DefaultAction,
+						OnDelete: DefaultAction,
+					},
+				},
+			},
+		},
+	}
+
+	testForeignKey(t, tests)
+}
+
+func TestAlterDropForeignKey(t *testing.T) {
+	tests := []testForeignKeyStruct{
+		{
+			`ALTER TABLE child DROP FOREIGN KEY fk_parent_id`,
```

Comment:
Should this work for unnamed foreign keys constraints too? How does MySQL do this?

Reply:
If you don't provide a name then MySQL generates a name for you, and you'll have to use that name to drop it.